proxy/src/transparency/client.rs

```diff
@@ -73,9 +73,13 @@ where
     B: tower_h2::Body + 'static,
 {
     /// Create a new `Client`, bound to a specific protocol (HTTP/1 or HTTP/2).
-    pub fn new(protocol: bind::Protocol, connect: C, executor: Handle) -> Self {
-        match protocol {
-            bind::Protocol::Http1 => {
+    pub fn new(protocol: &bind::Protocol,
```

Comment:
So this is where I mean that some state could be passed on that the `Client` needs to rewrite the URI with the `SO_ORIGINAL_DST` address as the authority.

Reply:
Oh, I get it now --- I didn't realise what you meant earlier, but I think I get it now, thanks!

Reply:
Hmm, I'm not actually sure if this will work --- making this change breaks the two unit tests for requests without hosts. I'll continue trying to make it work, though.